pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java

```diff
@@ -366,7 +372,15 @@ private RecordMetadata getRecordMetadata(String topic, TypedMessageBuilder<byte[
 
         TopicPartition tp = new TopicPartition(topic, partition);
         TypedMessageBuilderImpl<byte[]> mb = (TypedMessageBuilderImpl<byte[]>) msgBuilder;
-        return new RecordMetadata(tp, offset, 0L, mb.getPublishTime(), 0L, mb.hasKey() ? mb.getKey().length() : 0, size);
+
+        long publishTime = 0L;
+        try {
+            // there is no hasPublishTime() currently
+            publishTime = mb.getPublishTime();
+        } catch (IllegalStateException ise) {
+            logger.debug("could not get publish time");
```

Comment:
Why is the log level debug? I think `sendAsync` will set the publish time and `getPublishTime()` failure is an unexpected case so here the log level should be error.

Reply:
all messages that passed through LIneSplit example had this problem. 
Error logging was too noisy.
I didn't have time to investigate where it is expected to be filled in / why it was blank.

Reply:
@dlg99 good point. +1